ui/src/main/webapp/src/app/components/reports/application-index/application-index.component.ts

```diff
@@ -0,0 +1,223 @@
+import { NotificationService } from "../../../services/notification.service";
+import { ApplicationGroup } from "windup-services";
+import { AggregatedStatisticsService } from "./aggregated-statistics.service";
+
+@Component({
+    templateUrl: '/application-index.component.html',
+    styleUrls: ['/application-index.component.css']
+})
+export class ApplicationIndexComponent implements OnInit {
+
+    // color scheme for graphs on template
+    colorScheme = {
+        domain: ['#2E4272', '#553900', '#4F628E', '#162955', '#061539', '#AA8439', '#FFE3AA']
+    };
+
+    // both are used for getting graph and for report filter DTO
+    private execID: number;
+    private group: ApplicationGroup;
+
+    // aggregated statistics variables
+    globalPackageUseData: ChartStatistic[] = [];
+    categoriesMultiStats: any[] = [];
+    mandatoryMultiStats: any[] = [];
+
+    constructor(
+        private _router: Router,
+        private _activatedRoute: ActivatedRoute,
+        private _notificationService: NotificationService,
+        private _aggregatedStatsService: AggregatedStatisticsService,
+        private _http: Http
+    ) { }
+
+
+    ngOnInit(): void {
+        this._activatedRoute.params.subscribe(params => {
+            let executionId = parseInt(params['executionId']);
+            this.execID = executionId;
+
+
+            this._aggregatedStatsService.getAggregatedCategories(executionId).subscribe(
+                result => this.categoriesMultiStats = this.getCategoriesStats(this.calculateCategoryIncidents(result), this.calculateStoryPointsInCategories(result)),
+                error => {
+                    this._notificationService.error(utils.getErrorMessage(error))
+                    this._router.navigate(['']);
+                }
+            );
+            this._aggregatedStatsService.getMandatoryIncidents(executionId).subscribe(
+                result => this.mandatoryMultiStats = this.getCategoryDetailsStats(this.convertToChartStatistic(result)),
+                error => {
+                    this._notificationService.error(utils.getErrorMessage(error))
+                    this._router.navigate(['']);
+                }
+            );
+            this._aggregatedStatsService.getAggregatedJavaPackages(executionId).subscribe(
+                result => this.globalPackageUseData = this.convertPackagesToChartStatistic(result),
+                error => {
+                    this._notificationService.error(utils.getErrorMessage(error))
+                    this._router.navigate(['']);
+                }
+            );
+        });
+
+        this._activatedRoute.parent.parent.parent.data.subscribe((data: { applicationGroup: ApplicationGroup }) => {
+            this.group = data.applicationGroup;
+        });
+    }
+
+    /**
+     * Array of package statistics converted into ChartStatistic array for table and chart rendering
+     */
+    private convertPackagesToChartStatistic(statistics): ChartStatistic[] {
```

Comment:
I agree, I had some troubles with some types I used before, but I can try to redo this.